linker/shared/src/test/scala/org/scalajs/linker/testutils/IRAssertions.scala

```diff
@@ -0,0 +1,146 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker.testutils
+
+import scala.language.implicitConversions
+
+import scala.util.control.ControlThrowable
+
+import org.scalajs.ir.ClassKind
+import org.scalajs.ir.EntryPointsInfo
+import org.scalajs.ir.Definitions._
+import org.scalajs.ir.Traversers._
+import org.scalajs.ir.Trees._
+import org.scalajs.ir.Types._
+
+import org.scalajs.linker.standard._
+
+import org.junit.Assert._
+
+object IRAssertions {
+  implicit def classDefAssertions(classDef: ClassDef): ClassDefAssertions =
+    new ClassDefAssertions(classDef)
+
+  implicit def linkedClassAssertions(linkedClass: LinkedClass): LinkedClassAssertions =
+    new LinkedClassAssertions(linkedClass)
+
+  type Pat = PartialFunction[IRNode, Boolean]
+
+  private def patToTotal(pat: Pat): IRNode => Boolean =
+    node => pat.applyOrElse(node, (_: IRNode) => false)
+
+  abstract class AbstractIRNodeAssertions[T](selfNode: T) {
+    protected def newTraverser(f: IRNode => Unit): TestTraverser[T]
+
+    protected def newFinder(f: IRNode => Boolean): TestFinder[T] =
```

Comment:
Any reason this is not private?

Also, consider calling this `find` and basically put everything in here (including `patToTotal`).